src/image.rs

```diff
@@ -0,0 +1,271 @@
+use std::{collections::HashMap, env::var, thread::sleep, time::Duration};
+use testcontainers::{
+    core::{Container, Docker, Port, WaitForMessage},
+    Image,
+};
+
+// TODO: Switch to using 18081 port since --regtest is mainnet
+
+#[derive(Debug)]
+pub struct Monero {
+    tag: String,
+    args: Args,
+    ports: Option<Vec<Port>>,
+    entrypoint: Option<String>,
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct Args {
+    monerod: MonerodArgs,
+    wallet: WalletArgs,
+}
+
+#[derive(Debug, Clone)]
+pub struct MonerodArgs {
+    pub regtest: bool,
+    pub rpc_payment_allow_free_loopback: bool,
+    pub confirm_external_bind: bool,
+    pub non_interactive: bool,
+    pub no_igd: bool,
+    pub hide_my_port: bool,
+    pub rpc_bind_ip: String,
+    pub rpc_bind_port: u16,
+    pub fixed_difficulty: u32,
+    pub data_dir: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct WalletArgs {
+    pub disable_rpc_login: bool,
+    pub confirm_external_bind: bool,
+    pub wallet_dir: String,
+    pub rpc_bind_ip: String,
+    pub rpc_bind_port: u16,
+    pub daemon_address: String,
+    pub log_level: u32,
+}
+
+/// Sane defaults for a mainnet regtest instance.
```

Comment:
I am curious, what is a "mainnet regtest instance"?

Reply:
I'm confused about all this too. Seems for monero their networks also imply what branc/commit they are working off. `mainnet` and `stagenet` use release tagged commits. `testnet` uses tip of master branch. And, IIUC, `--regtest` (or regtest mode) uses same commit as mainnet and the same default settings i.e., port numbers but makes available some extra RPC commands (`generateblocks`) and allows one to set the mining difficulty with `--fixed-difficulty`. 

I will try to word this code comment better.